.github/workflows/deploy.yaml

```diff
@@ -5,13 +5,37 @@ on:
     branches:
       - 'master'
   workflow_dispatch:
-
+    inputs:
+        environment:
+          description: 'Miljø'
+          required: true
+          default: 'q2'
 env:
   IMAGE: docker.pkg.github.com/${{ github.repository }}/melosys-soknad-mottak:${{ github.sha }}
+  IMAGE_MANIFEST: https://docker.pkg.github.com/v2/${{ github.repository }}/melosys-soknad-mottak/manifests/${{ github.sha }}
 
 jobs:
-  build_and_deploy:
+  check_docker_image_exists:
+    name: Check if docker image exists in registry
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          echo "DOCKER_IMAGE_EXISTS_HTTP_STATUS=$(curl -X GET -s -o /dev/null -w "%{http_code}" ${{ env.IMAGE_MANIFEST }} -u ${{ github.actor }}:${{ secrets.GITHUB_TOKEN }})" >> $GITHUB_ENV
+      - id: set_output
+        run: |
+          if [ ${{ env.DOCKER_IMAGE_EXISTS_HTTP_STATUS }} -eq 200 ]
+          then
+            echo "::set-output name=exists::true"
+          else
+            echo "::set-output name=exists::false"
+          fi
+    outputs:
+      exists: ${{ steps.set_output.outputs.exists }}
+
+  build:
     runs-on: ubuntu-latest
+    needs: check_docker_image_exists
+    if: needs.check_docker_image_exists.outputs.exists != 'true'
```

Comment:
Kan du forklare hvorfor det er `needs.check_docker_image_exists.outputs.exists`? Jeg forventet enten `check_docker_image_exists.outputs.exists` eller `needs.outputs.exists`.

Reply:
`needs` er nødvendig for å referere til `job`'ene som `build` er avhengig av(kan være avhengig av flere).

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs

> To use job outputs in a dependent job, you can use the needs context.